mvcc/watcher.go

```diff
@@ -33,15 +34,17 @@ type FilterFunc func(e mvccpb.Event) bool
 
 type WatchStream interface {
 	// Watch creates a watcher. The watcher watches the events happening or
-	// happened on the given key or range [key, end) from the given startRev.
+	// happened on the given key or range [key, end) from the given startRev. If
+	// the ID is provided and non-zero, it will be used, otherwise a new ID
```

Comment:
it seems that -1 is the value, not non-zero.

Reply:
So, it actually is zero, because it doesn't look like the protobuf is set to have optional types (so, if the watcher ID is not present in the WatchRequest, it's zero). There is some kind of weird casing around 0 and -1 right now, since I didn't change the -1 return the function had previously.

Instead, I want to normalize around 0 and get rid of the -1 return in favor of a sentinel error.